### DISCUSSION TITLE
Saving WebArchive page from Safari

### DISCUSSION THREAD

**Discussion by Cromemco** [gildas-lormeau/SingleFile#1073] [General]
SingleFile for Safari on Mac doesn't seem to be able to save pages previously saved as WebArchive files. I have some old WA files from defunct sites that I want to convert to a more standard HTML format since Apple has a habit of abandoning their proprietary formats abruptly, like QTVR. But when I open a WA file on Safari then try to save it with SingleFile, I get a red oval on the SingleFile button, even through the page is rendered properly in the browser. I've tried separating everything using WebArchive Folderizer and WebArchive Extractor first, but then the pages don't render properly. Any chance this can be worked around? I would understand if you say it's too small a niche to spend time on.

**Comment by gildas-lormeau** [comment]
I confirm that it is impossible through the extension, and I can reproduce the issue you described. However, it might be possible through the [CLI tool](https://github.com/gildas-lormeau/single-file-cli) since [Playwright](https://playwright.dev/) supports WebKit. I need to do some tests...

**Reply by Cromemco** [reply]
Appreciate it. If you can't, you can't. It's not a huge thing and SingleFile is already a great extension.

**Reply by gildas-lormeau** [reply]
So, I added the support of webkit in single-file-cli and did a test with a simple webarchive file I created in Safari. Unfortunately, that didn't work.
